Student Sketches and Answers.jl

```diff
@@ -27,238 +26,320 @@ begin
+#     sy_data_2 = DataFrame(uid = unique(a_df.uid), timestamp = Vector{Vector{String}}(undef, length(unique(a_df.uid))), y_vector = Vector{Vector{Vector{Float64}}}(undef, length(unique(a_df.uid))))
+
+
+#     function update_sy_vector(student_answers,qid, sketch_data, sy_data, student, 								current_sketch_y_vector = [], current_timestamp = [])
+# 		# println(student_answers)
+# 		if(size(student_answers,1)==2 || student_answers.q_id[1]==110)
+# 			end_timestamp_1 = student_answers[student_answers.q_id .== 110, 													:].timestamp[1]
+# 		else
+# 			end_timestamp_1 = student_answers[student_answers.q_id .== 111, 													:].timestamp[1]
+# 		end
+#         if qid == 110
+#             	animation_data = filter(row -> row.timestamp <= end_timestamp_1, 											sketch_data)
+#             	current_sketch_y_vector = zeros(22)
+#         else
+# 			if(size(student_answers,1)==2 || student_answers.q_id[1]==111)
+# 	            end_timestamp_2 = student_answers[student_answers.q_id .== 111, 																	:].timestamp[1]
+# 			else
+# 				end_timestamp_2=sketch_data.timestamp[length(sketch_data.timestamp)]
+# 			end
+# 				animation_data = filter(row -> (end_timestamp_1 <= row.timestamp) && 					(row.timestamp <= end_timestamp_2), sketch_data)
+#         end
+
+#         x = animation_data.xcor
+#         y = animation_data.ycor
+#         t = animation_data.timestamp
+
+#         for i in 1:size(animation_data, 1)
+#             current_sketch_y_vector[x[i] + 1] = y[i]
+#             push!(sy_data.timestamp[student], t[i])
+#             push!(sy_data.y_vector[student], copy(current_sketch_y_vector))
+#         end
+#     end
+
+#     for i in 1:length(unique(a_df.uid))
+# 		# println(unique(a_df.uid))
+# 		if unique(a_df.uid)[i]==73 || unique(a_df.uid)[i]==87 ||unique(a_df.uid)[i]==106 || unique(a_df.uid)[i]==100 || unique(a_df.uid)[i]==104
+# 			continue
+# 		end
+# 	    student_answers = a_df[a_df.uid .== unique(a_df.uid)[i], :]
+# 	    sketch_data = filter(row -> row.uid == unique(a_df.uid)[i], s_df)
+# 		# if(i==2)
+# 		# 	println(student_answers)
+# 		# 	println(sketch_data)
+# 		# end
+#         sy_data_1.timestamp[i] = []
+#         sy_data_1.y_vector[i] = []
+#         update_sy_vector(student_answers,110, sketch_data, sy_data_1, i)
+
+#         sy_data_2.timestamp[i] = []
+#         sy_data_2.y_vector[i] = []
+# 		# println(sy_data_1.y_vector[student][length(sy_data_1.y_vector[student])-1])
+#         update_sy_vector(student_answers,111, sketch_data, sy_data_2, i, sy_data_1.y_vector[i][length(sy_data_1.y_vector[i])])
+		
+#     end
+
+#     return sy_data_1, sy_data_2
+# end
+
+
+# ╔═╡ a7751416-34d8-4cf5-a229-78a3a074c6c1
+function PLOT(sy_data,t)
+	# println(sy_data)
+	begin_stamp=sy_data.timestamp[1]
+end_stamp=sy_data.timestamp[length(sy_data.timestamp)]
```

Comment:
You can just use `end` to get the last row instead of `length(...)`, like this: `current_data.y_vector[end]`. There are a few places you use `length` this way to get the last element of a vector. Replace them all with `end`.